rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -330,21 +286,22 @@ object Reduce {
                   case None => Applicative[M].pure(Left((target, caseRem)))
                   case Some(freeMap) => {
                     val newEnv: Env[Par] = addToEnv(env, freeMap, singleCase.freeCount)
-                    eval(singleCase.source.get)(newEnv).map(Right(_))
+                    eval(singleCase.source)(newEnv).map(Right(_))
                   }
                 }
               }
           }
         }
+
         FlatMap[M].tailRecM[Tuple2[Par, Seq[MatchCase]], Unit]((target, cases))(firstMatchM)
       }
 
       for {
-        evaledTarget <- evalExpr(mat.target.get)
-        // TODO(kyle): Make the matcher accept an environment, instead of
-        // substituting it.
+        evaledTarget <- evalExpr(mat.target)
+        // TODO(kyle): Make the matcher accept an environment, instead of substituting it.
         substTarget <- substitutePar[M].substitute(evaledTarget)(0, env)
         _           <- firstMatch(substTarget, mat.cases)
+        _           <- costAccountingAlg.charge(MATCH_EVAL_COST)
```

Comment:
@guardbotmk3 I've added constant cost for each term evaluation (`MATCH_EVAL_COST`, `SEND_EVAL_COST`, `METHOD_CALL_COST` etc) but I am not convinced yet about it.